### PR TITLE
Return ALL types when selecting possible values or creating new entities

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

We have a few places where people can see / select from entity types:
1. In the left sidebar, they can see a list of types
2. When creating a new entity, they can select a type to assign it
3. When creating an entity type property, they can select another entity type as the permitted value

Previously, the types shown in these lists were the types belonging to an account AND any types that were 'in use' in that account, i.e. there were entities of that type in the account.

This meant that it was impossible for users to see / use types which were neither of their account or already in use by their account.

This PR starts returning ALL types for use in those lists, so that users can do that. As the API will be rebuilt for the new type system, this involves some temporary hacks (see known issues).

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202566167655642) _(internal)_

## ⚠️ Known issues

- there is no `aggregateEntityTypes` API or db query, so this PR adds an `includeAllTypes` option to `getAccountEntityTypes`. this is obviously inappropriate but these functions will all be revamped with the new API so it isn't worth giving them any more thought now
- there are TWO hooks that allow users to retrieve entity types. these need to be consolidated

## ❓ How to test this?

1.  See if you can see all types in the left sidebar whichever account you choose

## 📹 Demo

Screenshot 2022-07-21 at 09 37 53

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202500568584700

### CODE REVIEW

File: packages/hash/integration/src/mockData/index.ts [hunk 1]
```diff
@@ -65,11 +64,6 @@ void (async () => {
     createOrgs(db),
   ]);
 
-  await createEntityTypes(db)([
```
Comment (thehabbos007) [inline packages/hash/integration/src/mockData/index.ts:68]: Were these types not used for any of the integration tests? It's nice we can get rid of it, but interesting we never used this

**Review by thehabbos007 — DISMISSED** [review]
The changes look good, i just have some comments :)

File: packages/hash/integration/src/mockData/index.ts (hunk 1 above)
Comment (CiaranMn) [reply packages/hash/integration/src/mockData/index.ts:68]: No – we had an example `Task` type which we put in everyone's namespace. This meant that a list of 'all types' contained 'Task' multiple times, which was pretty silly, and it doesn't seem important to have this dummy data for testing (people can create their own type if they want one).

File: packages/hash/api/src/graphql/typeDefs/entityType.typedef.ts [hunk 2]
```diff
@@ -17,12 +17,17 @@ export const entityTypeTypedef = gql`
...
-    Optionally include types in use by the account, but belonging to other accounts.
+    Optionally include:
+    - types in use by the account, but belonging to other accounts.
+    - ALL types.
+    TODO: replace this with a new aggregateEntityTypes query instead
     """
     getAccountEntityTypes(
       accountId: ID!
+      includeAllTypes: Boolean = false
       includeOtherTypesInUse: Boolean = false
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply packages/hash/api/src/graphql/typeDefs/entityType.typedef.ts:31]: I agree this query is now somewhat badly named but I expect we'll replace it entirely before anyone else has any reason to use it, so I'd just leave it here until we need to make any further changes here (which if anything will be me if we need changes before the graph API is ready)

File: packages/hash/frontend/src/components/hooks/useGetAllEntityTypes.ts [hunk 3]
```diff
@@ -5,18 +5,19 @@ import {
 } from "../../graphql/apiTypes.gen";
 import { getAccountEntityTypes } from "../../graphql/queries/account.queries";
 
-export const useAccountEntityTypes = (
-  accountId: string,
-  includeOtherTypesInUse?: boolean,
-) => {
+/**
+ * @todo consolidate this and useBlockProtocolAggregateEntityTypes
```
Comment (benwerner01) [inline packages/hash/frontend/src/components/hooks/useGetAllEntityTypes.ts:9]: I think we might be able to just outright delete the `useGetAllEntityTypes` hook, assuming `useBlockProtocolAggregateEntityTypes` will let us aggregate the entity types in a users account. Appreciate this is probably what you mean with "consolidate"

**Review by benwerner01 — APPROVED** [review]
Thanks Ciaran, this is working as intended locally for me and looks good to me 👍

**Pull request merged**